libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -178,13 +229,119 @@ def test_custom_loader_factory_with_configurations(
     create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
     expected_custom_csv_documents_with_columns: list[Document],
 ) -> None:
-    def csv_loader_factory(file_path: str) -> CustomCSVLoader:
-        return CustomCSVLoader(
-            file_path=file_path,
-            content_columns=["col1"],
-        )
-
     loader = create_azure_blob_storage_loader(
-        blob_names="csv_file.csv", loader_factory=csv_loader_factory
+        blob_names="csv_file.csv", loader_factory=get_first_column_csv_loader
     )
     assert list(loader.lazy_load()) == expected_custom_csv_documents_with_columns
+
+
+async def test_alazy_load(
+    account_url: str,
+    container_name: str,
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+) -> None:
+    loader = create_azure_blob_storage_loader()
+    expected_document_list = get_expected_documents(
+        get_test_blobs(), account_url, container_name
+    )
+    assert [doc async for doc in loader.alazy_load()] == expected_document_list
+
+
+@pytest.mark.parametrize(
+    "blob_names",
+    [
+        "text_file.txt",
+        ["text_file.txt", "json_file.json"],
+    ],
+)
+async def test_alazy_load_with_blob_names(
+    blob_names: Union[str, Iterable[str]],
+    account_url: str,
+    container_name: str,
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+    async_mock_container_client: Tuple[AsyncMock, AsyncMock],
+) -> None:
+    _, async_mock_client = async_mock_container_client
+    loader = create_azure_blob_storage_loader(blob_names=blob_names)
+    expected_documents_list = get_expected_documents(
+        get_test_blobs(blob_names), account_url, container_name
+    )
+    assert [doc async for doc in loader.alazy_load()] == expected_documents_list
+    assert async_mock_client.list_blob_names.call_count == 0
+
+
+async def test_get_async_blob_client(
+    create_azure_blob_storage_loader: Callable[..., AzureBlobStorageLoader],
+    async_mock_container_client: Tuple[AsyncMock, AsyncMock],
+) -> None:
+    _, async_mock_client = async_mock_container_client
+
+    async def updated_prefix_blob_list(**kwargs: Any) -> AsyncIterator[str]:
+        yield "text_file.txt"
+
+    async_mock_client.list_blob_names.side_effect = updated_prefix_blob_list
```

Comment:
In our `async_mock_container_client`, could we just proxy the `prefix` kwarg into `get_test_blobs()` to avoid needing to define this logic?